src/System.Linq.Expressions/src/System/Linq/Expressions/Compiler/LambdaCompiler.Address.cs

```diff
@@ -109,7 +109,16 @@ private void AddressOf(ParameterExpression node, Type type)
             }
             else
             {
-                EmitExpressionAddress(node, type);
+                // NB: Stack spilling can generate ref locals.
+
+                if (node.Type == type.MakeByRefType())
```

Comment:
Would it make sense to avoid "MakeByRefType" here. Can we check "IsByRef" and then compare underlying type?